src/lib/data-root.ts

```
@@ -0,0 +1,287 @@
+        if (data !== undefined) {
+          try {
+            data.stream.destroy();
+          } catch (error: any) {
+            log.error('Error destroying data stream', {
+              message: error?.message,
+              stack: error?.stack,
+            });
+          }
+        }
+      }
+    });
```

Biome

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)